### PR TITLE
Add .packmindignore support for CLI lint command

### PR DESCRIPTION
**Pull request by cteyton**
## Explanation

Add support for `.packmindignore` files in the CLI lint command. This allows users to exclude additional files/folders from analysis without modifying CLI source code, similar to `.gitignore`.

The `.packmindignore` file supports:
- One pattern per line (same glob syntax as existing `ListFiles.matchesGlobPattern()`)
- Comments (`#` lines) and empty lines are ignored
- Hierarchical lookup: walks up from linted directory to git repo root, merging all found `.packmindignore` files
- Patterns are additive on top of existing hardcoded excludes (`node_modules`, `dist`, `.min.`, `.map.`, `.git`)

## Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Improvement/Enhancement
- [ ] Refactoring
- [ ] Documentation
- [ ] Breaking change

## Affected Components

- Domain packages affected: `packmind-cli`
- Frontend / Backend / Both: CLI only
- Breaking changes (if any): None

## Testing

- [x] Unit tests added/updated
- [ ] Integration tests added/updated
- [ ] Manual testing completed
- [x] Test coverage maintained or improved

**Test Details:**
- `PackmindIgnoreReader.spec.ts`: 7 tests covering parsing, comment stripping, whitespace trimming, multi-level merging, and edge cases
- `lintHandler.spec.ts`: 4 new tests verifying ignore patterns are read and passed to both `lintFilesFromConfig` and `lintFilesAgainstRule`
- All 1747 existing tests continue to pass

## TODO List

- [x] CHANGELOG Updated
- [x] Documentation Updated

## Reviewer Notes

- The `lintHandler` was refactored to consolidate 3 separate `tryGetGitRepositoryRoot` calls into 1, which is a minor improvement alongside the feature
- `.packmindignore` patterns apply to all lint modes: full scan, `--changed-files`, `--changed-lines`, and `--rule`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by cteyton** [comment]
@greptile update your review

File: apps/cli/src/infra/commands/lintHandler.ts [hunk 1]
```diff
@@ -78,15 +81,19 @@ export async function lintHandler(
...
+  if (diff && !gitRoot) {
+    throw new Error(
+      'The --changed-files and --changed-lines options require the project to be in a Git repository',
+    );
   }
 
+  const ignorePatterns = await packmindIgnoreReader.readIgnorePatterns(
+    absolutePath,
+    gitRoot,
+  );
```
[the remark this answers is not in the document]
Comment (cteyton) [reply apps/cli/src/infra/commands/lintHandler.ts]: @greptile we don't support running the "lint" command against a single file, just a directory.

File: apps/cli/src/application/useCases/LintFilesFromConfigUseCase.ts [hunk 2]
```diff
@@ -200,12 +201,17 @@ export class LintFilesFromConfigUseCase implements ILintFilesFromConfig {
...
+    }
+
     let files = isFile
       ? [{ path: absoluteUserPath }]
       : await this.services.listFiles.listFilesInDirectory(
           absoluteUserPath,
           [],
-          ['node_modules', 'dist', '.min.', '.map.', '.git'],
+          excludes,
         );
```
[the remark this answers is not in the document]
Comment (cteyton) [reply apps/cli/src/application/useCases/LintFilesFromConfigUseCase.ts:215]: @greptile We'll handle single file analysis in a separate issue so let's ignore it.

File: apps/cli/src/application/services/ListFiles.ts [hunk 3]
```diff
@@ -3,6 +3,14 @@ import * as path from 'path';
...
 import { FileResult, IListFiles } from '../../domain/services/IListFiles';
 
+export const DEFAULT_EXCLUDES = [
+  'node_modules',
+  'dist',
+  '*.min.*',
+  '*.map',
+  '.git',
+];
+
```
[the remark this answers is not in the document]
Comment (cteyton) [reply apps/cli/src/application/services/ListFiles.ts:13]: You're right, I need to fix this.

File: apps/doc/linter/linter.mdx [hunk 4]
```diff
@@ -184,6 +184,42 @@ When creating rules that apply to multiple languages:
...
+
+# Ignore legacy JavaScript files in src
+src/*.legacy.js
+```
+
+<Note>
+  The following paths are **always excluded** by default, even without a
+  `.packmindignore` file: `node_modules`, `dist`, `.git`, `.min.*` files, and
+  `.map.*` files.
+</Note>
```
[the remark this answers is not in the document]
Comment (cteyton) [reply apps/doc/linter/linter.mdx:221]: @greptile do you confirm this has been fixed ?

File: apps/cli/src/application/services/ListFiles.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (cteyton) [reply apps/cli/src/application/services/ListFiles.ts:13]: @greptile do you confirm this has been fixed ?
